### ISSUE TITLE
[YSQL] [Perf] [TPCC]  Multiple exceptions logged on TPCC client logs, in the execution phase.

### ISSUE THREAD

**Issue by sonalsagarwal** [yugabyte/yugabyte-db#12569]
Jira Link: [[DB-432]](https://yugabyte.atlassian.net/browse/DB-432)
### Description

The following error message for  foreign key constraint violation is logged multiple times in tpcc client logs. This is observed in the perf regression for 2.13.3.0-b59 onwards.
```
WARN  - The DBMS rejected the transaction without an error code
org.postgresql.util.PSQLException: ERROR: insert or update on table \"order_line\" violates foreign key constraint \"ol_fkey_s\"
  Detail: Key (ol_supply_w_id, ol_i_id)=(726, 35927) is not present in table \"stock\".
     at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2468)
     at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2211)
     at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:309)
     at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:446)
     at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:370)
     at org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:149)
     at org.postgresql.jdbc.PgPreparedStatement.execute(PgPreparedStatement.java:138)
     at com.zaxxer.hikari.pool.ProxyPreparedStatement.execute(ProxyPreparedStatement.java:44)
     at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.execute(HikariProxyPreparedStatement.java)
     at com.oltpbenchmark.jdbc.InstrumentedPreparedStatement.execute(InstrumentedPreparedStatement.java:63)
     at com.oltpbenchmark.benchmarks.tpcc.procedures.NewOrder.insertOrderLines(NewOrder.java:667)
     at com.oltpbenchmark.benchmarks.tpcc.procedures.NewOrder.newOrderTransaction(NewOrder.java:366)
     at com.oltpbenchmark.benchmarks.tpcc.procedures.NewOrder.run(NewOrder.java:258)
     at com.oltpbenchmark.api.Worker.executeWork(Worker.java:609)
     at com.oltpbenchmark.api.Worker.doWork(Worker.java:506)
     at com.oltpbenchmark.api.Worker.run(Worker.java:368)
     at java.lang.Thread.run(Thread.java:750)\n08:47:55,065 (Worker.java:545)
```

[DB-432]: https://yugabyte.atlassian.net/browse/DB-432?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by sonalsagarwal** [comment]
Tablet splitting is enabled in the perf regression runs. I have also seen these stack traces when tablet splitting is disabled.

**Comment by d-uspenskiy** [comment]
Fixed in context of #12648
